### PR TITLE
fix: Linter fixes for plugins/aggregators/[a-z]*

### PR DESCRIPTION
**Pull request by zak-pawel**
Following findings were fixed:
```
plugins/aggregators/basicstats/basicstats.go:132:5        revive    unexported-naming: the symbol M2 is local, its name should start with a lowercase letter
plugins/aggregators/derivative/derivative_test.go:43:17   errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:69:17   errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:91:17   errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:133:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:166:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:198:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:223:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:263:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:312:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:335:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/derivative/derivative_test.go:359:17  errcheck  Error return value of `derivative.Init` is not checked
plugins/aggregators/merge/merge_test.go:232:13            errcheck  Error return value of `merger.Init` is not checked
plugins/aggregators/merge/merge_test.go:244:13            errcheck  Error return value of `merger.Init` is not checked
plugins/aggregators/quantile/algorithms.go:52:2           revive    unexported-naming: the symbol N is local, its name should start with a lowercase letter
plugins/aggregators/quantile/algorithms.go:98:2           revive    unexported-naming: the symbol N is local, its name should start with a lowercase letter
```

### CODE REVIEW

File: plugins/aggregators/histogram/histogram_test.go [hunk 1]
```diff
@@ -83,7 +84,7 @@ func TestHistogram(t *testing.T) {
 	histogram.Push(acc)
 
 	if len(acc.Metrics) != 6 {
-		assert.Fail(t, "Incorrect number of metrics")
+		require.Fail(t, "Incorrect number of metrics")
 	}
```
Comment (srebhan) [inline plugins/aggregators/histogram/histogram_test.go:88]: Can we please convert this whole part to
```golang
	require.Len(t, acc.Metrics, 6, "Incorrect number of metrics")
```
same for the other occurrences below.

**Review by srebhan — COMMENTED** [review]
Thanks @zak-pawel! Just one comment to use `require.Len` instead of `if len... require.Fail`...

File: plugins/aggregators/histogram/histogram_test.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/aggregators/histogram/histogram_test.go]: Done.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for fixing those issues!

**Review by sspaink — APPROVED** [review]

**Pull request merged**